src/app.py

```diff
@@ -91,14 +91,26 @@ def query(
             for rule in rules
         ]
 
-    active_agents = list(db.get_active_agents().keys())
+    active_agents = db.get_active_agents()
+
+    for agent, agent_spec in active_agents.items():
+        missing = set(data.required_plugins).difference(
```

Comment:
technically sometimes you might also want to ignore the agent :thinking:. But error makes sense as a default I guess.